### ISSUE TITLE
Configuration Reload doesn't work

### ISSUE THREAD

**Issue by vpolischuk** [jenkinsci/configuration-as-code-plugin#1523]
- [x] Jenkins version: 1.45

- [x] Plugin version: 2.249.2

- [x] OS: Debian GNU/Linux 10 (buster)

### Description

I added some credentials/jobs(via job dsl) and tried reload jenkins via web ui (jenkins >> Configuration as Code >> Reload existing configuration)

Expected behavior:
jcasc will create new credentials/jobs

Actual behavior:
no new credentials/jobs, without any errors in jenkins log

Trying reload via http POST to JENKINS_URL/reload-configuration-as-code (with token) - same effect.

New credentials/jobs added only after restart.

**Comment by timblaktu** [comment]
@vpolischuk is CasC reload function supposed to work without a restart of jenkins service? How is reload expected to behave when there are running pipleine jobs?

**Comment by Rubenmp** [comment]
This issue is happening in a fresh installation of Jenkins (without running jobs) using 'Replace configuration source with:'. It throws a generic error "A problem occurred while processing the request."

I tested it using this [url](https://gist.githubusercontent.com/Rubenmp/c40be9a8ff7e99e3f3444a448a1b0c49/raw/d2e8b176d5127f181529076f6baaab4580bc3dd2/JenkinsCasc):

**Comment by timja** [comment]
check the system logs

**Comment by brandon-fryslie** [comment]
I've got a question about how "safe" it is to reload the JCasC configuration on-the-fly when jobs are running.  Are there known issues where it will break running jobs?  We aren't planning on changing any job configuration (we already use the job-dsl plugin for that), but we want to be able to change Jenkins configuration.  I'm trying to find any information on potential issues that might cause

**Comment by timja** [comment]
The new configuration will only be applied to new jobs afaik so should be safe, we've never hit an issue with it

---

please open a new issue for anything further
